test/pdevs_simulator_test.cpp

```diff
@@ -146,6 +148,89 @@ BOOST_AUTO_TEST_CASE( accumulator_simulation_throws_test ){
     BOOST_CHECK(s.next()==std::numeric_limits<float>::infinity());
 }
 
+BOOST_AUTO_TEST_CASE( accumulator_model_dynamic_simulation_test )
+{
+    //construct a simulator for an accumulator
+    using simulator_t=cadmium::engine::dynamic_simulator<int_accumulator, float, cadmium::logger::not_logger>;
+    simulator_t s;
+    s.init(0.0f);
+
+    BOOST_CHECK(s.next()==std::numeric_limits<float>::infinity());
+
+    using input_ports=int_accumulator<float>::input_ports;
+    using in_bags_type=typename cadmium::make_message_bags<input_ports>::type;
+
+    //crate the map for sending/receiving the messages
+    cadmium::message_bag<int_accumulator_defs::add> bag_0;
+    cadmium::message_bag<int_accumulator_defs::reset> bag_1;
+    cadmium::message_bag<int_accumulator_defs::sum> output;
+
+    bag_0.messages.assign(std::initializer_list<int>{1, 2, 3, 4});
+    bag_1.messages.clear();
+
+    cadmium::dynamic_message_bags input_bags;
+    cadmium::dynamic_message_bags empty_input = cadmium::modeling::create_empty_dynamic_message_bags<in_bags_type>();
+
+    input_bags[typeid(bag_0)] = bag_0;
+    input_bags[typeid(bag_1)] = bag_1;
+
+    //advance simulator
+    s.inbox(input_bags);
+    s.advance_simulation(3.0f);
```

Comment:
collect_outputs is expected to be called before advance_simulation always. Even if we expect it not to do anything in this particular case, it may worth keeping it in the test to be sure it does not interfere.

Reply:
I've just translated the test from the one you did for the simulator, should I follow this suggestion for the simulation test you did too?

Reply:
Sure.